libs/SalesforceSDK/src/com/salesforce/androidsdk/config/LoginServerManager.java

```diff
@@ -177,6 +179,14 @@ public void useSandbox() {
 	 * @param url Server URL.
 	 */
 	public void addCustomLoginServer(String name, String url) {
+		// Prevent duplicate servers.
+		for (LoginServer existingServer : getLoginServers()) {
+			if (name.equals(existingServer.name) && url.equals(existingServer.url)) {
+				setSelectedLoginServer(existingServer);
+				return;
+			}
+		}
```

Comment:
It did not make sense to check for this in the UI code.  If we did, we would then have to surface some kind of error message.  Simply not adding the duplicate and selecting the original in the data source of truth seemed like an elegant solution.  

And I purposefully did not use our equals method because I wanted to ignore `isCustom`.